### PR TITLE
Add MVVM Toolkit generators docs

### PR DESCRIPTION
**Pull request by Sergio0694**
This PR adds docs for all the source generators in the MVVM Toolkit.

### CODE REVIEW

**Comment by PRMerger4** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 Changed Files | This PR contains more than 10 changed files.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger17** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 Changed Files | This PR contains more than 10 changed files.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by michael-hawker** [comment]
@Sergio0694 I think images can only be referenced within our repo as it's a relative path from the current directory of the md file and the 'root' is our repo of `CommunityToolkit`? So, if you want to reference something outside our docs repo, I think you'll have to use the full URL? Unless there's some other magic we need to make this work for the docs path guidance? @DCtheGeek?

**Comment by Sergio0694** [comment]
> "if you want to reference something outside our docs repo, I think you'll have to use the full URL?"

But then how is that working for relative links for other APIs? 🤔

**Comment by PRMerger17** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 Changed Files | This PR contains more than 10 changed files.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Image File | This PR added or updated an image file(s).

**Comment by PRMerger4** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 Changed Files | This PR contains more than 10 changed files.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Image File | This PR added or updated an image file(s).
 Index File | This PR contains a change to the index file.

File: docs/mvvm/generators/INotifyPropertyChanged.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Using any of these attributes is pretty straightforward: just add them to a [partial class](/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods) and all the code from the corresponding types will automatically be generated into that type. For instance, consider this:
+
+```csharp
+[INotifyPropertyChanged]
+public partial class MyViewModel : SomeOtherType
+{    
+}
+```
+
+This will generate a complete `INotifyPropertyChanged` implementation into the `MyViewModel` type, complete with additional helpers (such as `SetProperty`) that can be used to reduce verbosity. Here is a brief summary of the various attributes:
```
Comment (michael-hawker) [inline docs/mvvm/generators/INotifyPropertyChanged.md:30]: This block is a bit hard to parse visually with all the blocks:

image

At least in certain wrappings. Maybe don't fence all the things or make the starting ones bold, not sure the best way to resolve?

**Review by michael-hawker — COMMENTED** [review]
Looking good!

File: docs/mvvm/generators/RelayCommand.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+{
+    User user = await userService.GetCurrentUserAsync(token);
+
+    Console.WriteLine($"Hello {user.Name}!");
+}
+```
+
+In this case, the `try/catch` is not needed, as exceptions will not crash the app anymore. Note that this will also cause other unrelated exceptions to not be rethrown automatically, so you should carefully decide how to approach each individual scenario and configure the rest of the code appropriately.
+
+## Cancel commands for asynchronous operations
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/RelayCommand.md:153]: This has to be after the task scheduler exception flow paragraph, as we need to explain users why the new flag is causing this method to not crash the app anymore.

File: docs/mvvm/generators/INotifyPropertyChanged.md (hunk 1 above)
Comment (Sergio0694) [reply docs/mvvm/generators/INotifyPropertyChanged.md:33]: I've changed the titles to be links, which also makes them blue so they stand out more 🙂

File: docs/mvvm/generators/RelayCommand.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    {
+    }
+}
+```
+
+Will result in the generated command passing a token to the wrapped method. This allows consumers to just call [`IAsyncRelayCommand.Cancel`](/dotnet/api/communitytoolkit.mvvm.input.iasyncrelaycommand.cancel) to signal that token, and to allow pending operations to be stopped correctly.
+
+## Enabling and disabling commands
+
+It is often useful to be able to disable commands, and to then later on invalidate their state and have them check again whether they can be executed or not. In order to support this, the `RelayCommand` attribute exposes the `CanExecute` property, which can be used to indicate a target member to use to evaluate whether a command can be executed:
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/RelayCommand.md]: Changed to explicitly mention methods and properties 🙂

File: docs/mvvm/generators/Overview.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+```csharp
+[RelayCommand]
+private void SayHello()
+{
+    Console.WriteLine("Hello");
+}
+```
+
+With the new MVVM source generators, all of this is possible, and much more! 🙌
+
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/Overview.md:61]: Added a note section 🙂

**Comment by PRMerger-2** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 Changed Files | This PR contains more than 10 changed files.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Image File | This PR added or updated an image file(s).
 Index File | This PR contains a change to the index file.

File: docs/mvvm/generators/ObservableProperty.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    {
+        if (SetProperty(ref name, value))
+        {
+            ValidateProperty(value, "Value2");
+        }
+    }
+}
+```
+
+That generated `ValidateProperty` call will then validate the property and update the state of the `ObservableValidator` object, so that UI components can react to it and display any validation errors appropriately.
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/ObservableProperty.md:169]: Added a note section 🙂

File: docs/mvvm/generators/ObservableProperty.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    Console.WriteLine($"Name is about to change to {value}");
+}
+
+partial void OnNameChanged(string? value)
+{
+    Console.WriteLine($"Name has changed to {value}");
+}
+```
+
+You're free to only implement one of these two methods, or neither. If they are not implemented (or if only one is), the entire call(s) will just be removed by the compiler, so there will be no performance hit at all for cases where this additional functionality is not required.
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/ObservableProperty.md:85]: They don't have to be `private`, in fact, they _can't_ be private 😄

File: docs/mvvm/generators/ObservableProperty.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+author: Sergio0694
+description: An attribute to generate observable properties
+keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, mvvm, componentmodel, property changed, notification, binding, net core, net standard
+dev_langs:
+  - csharp
+---
+
+# ObservableProperty attribute
+
+The [`ObservableProperty`](/dotnet/api/communitytoolkit.mvvm.componentmodel.ObservablePropertyAttribute) type is an attribute that allows generating observable properties from annotated fields. Its purpose is to greatly reduce the amount of boilerplate that is needed to define observable properties. In order to work, annotated fields need to be in a [partial class](/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods) with the necessary `INotifyPropertyChanged` infrastructure.
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/ObservableProperty.md]: Moved to a note section 🙂

File: docs/mvvm/generators/Overview.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+These docs will go over exactly what features are included with the MVVM generators and how to use them.
+
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/Overview.md:72]: Done 😄

File: docs/mvvm/generators/Overview.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+You could just have a simple annotated field to express the same?
+
+```csharp
+[ObservableProperty]
+private string? name;
+```
+
+What about creating a command:
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/Overview.md:37]: Added links 👍

File: docs/mvvm/generators/Overview.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+Starting with version 8.0, the MVVM Toolkit includes brand new Roslyn source generators that will help greatly reduce boilerplate when writing code using the MVVM architecture. They can simplify scenarios where you need to setup observable properties, commands and more. If you're not familiar with source generators, you can read more about them [here](/dotnet/csharp/roslyn-sdk/source-generators-overview). This is a simplified view of how they work:
+
+![Roslyn source generators](../../images/source-generator-visualization.png "A diagram showing the architecture of Roslyn source generators, and the fact they generate code based on user code and add that to the existing compilation")
+
+This means that as you're writing code, the MVVM Toolkit generator will now take care of generating additional code for you behind the scenes, so you don't have to worry about it. This code will then be compiled and included in your application, so the end result is exactly the same as if you had written all that extra code manually, but without having to do all of that extra work! 🎉
+
+For instance, wouldn't it be great if instead of having to setup an observable property normally:
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply docs/mvvm/generators/Overview.md:18]: Added links here too 👍

**Comment by PRMerger19** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 Changed Files | This PR contains more than 10 changed files.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Image File | This PR added or updated an image file(s).
 Index File | This PR contains a change to the index file.
